File: knowledge/src/main/java/com/ayvytr/knowledge/presenter/GankHistoryContentPresenter.java
package com.ayvytr.knowledge.presenter;

import com.ayvytr.commonlibrary.BaseObserver;
import com.ayvytr.commonlibrary.bean.GankHistoryContent;
import com.ayvytr.knowledge.R;
import com.ayvytr.knowledge.contract.GankHistoryContentContract;
import com.ayvytr.knowledge.model.GankHistoryContentModel;
import com.ayvytr.mvp.BasePresenter;
import com.ayvytr.mvp.RxUtils;

/**
 * @author admin
 */
public class GankHistoryContentPresenter
        extends BasePresenter<GankHistoryContentContract.Model, GankHistoryContentContract.View> {
    public GankHistoryContentPresenter(GankHistoryContentContract.View view) {
        super(new GankHistoryContentModel(), view);
    }

    public void requestGankByDate(String date) {
        String[] split = date.split("-");
        mModel.getGankByDate(split[0], split[1], split[2])
              .compose(RxUtils.<GankHistoryContent>applySchedulers(mView))
              .compose(RxUtils.<GankHistoryContent>bindToLifecycle(mView))
              .subscribe(new BaseObserver<GankHistoryContent>() {
                  @Override
                  public void onNext(GankHistoryContent gankHistoryContent) {
                      if(gankHistoryContent.isError()) {
                          mView.showMessage(R.string.request_data_error);
                      } else {
                          mView.showGankByDate(gankHistoryContent);
                      }
                  }
              });
    }
}

File: knowledge/src/main/java/com/ayvytr/knowledge/presenter/GankHistoryContentPresenter.kt
package com.ayvytr.knowledge.presenter

import com.ayvytr.commonlibrary.BaseObserver
import com.ayvytr.commonlibrary.bean.GankHistoryContent
import com.ayvytr.knowledge.R
import com.ayvytr.knowledge.contract.GankHistoryContentContract
import com.ayvytr.knowledge.model.GankHistoryContentModel
import com.ayvytr.mvp.BasePresenter
import com.ayvytr.rxlifecycle.RxUtils

/**
 * @author admin
 */
class GankHistoryContentPresenter(view: GankHistoryContentContract.View) :
    BasePresenter<GankHistoryContentContract.Model, GankHistoryContentContract.View>(GankHistoryContentModel(), view) {

    fun requestGankByDate(date: String) {
        val split = date.split("-".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        mModel.getGankByDate(split[0], split[1], split[2])
            .compose(RxUtils.ofDefault(mView))
            .compose(RxUtils.bindToLifecycle(mView))
            .subscribe(object : BaseObserver<GankHistoryContent>() {
                override fun onNext(gankHistoryContent: GankHistoryContent) {
                    if (gankHistoryContent.isError) {
                        mView.showError(R.string.request_data_error)
                    } else {
                        mView.showGankByDate(gankHistoryContent)
                    }
                }
            })
    }
}
